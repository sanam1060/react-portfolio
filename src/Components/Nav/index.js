function Nav(props) {
  return;
  <nav className="navbar navbar-expand-md bg-light border border-secondary border-opacity-25">
    <div className="container">
      <a data-testid="navBarHeader" className="navbar-brand fw-bold" href="/">
        Sanam Ganjian
      </a>
      <button
        className="navbar-toggler"
        type="button"
        data-bs-toggle="collapse"
        data-bs-target="#navbarNav"
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon"></span>
      </button>
      <div
        className="collapse navbar-collapse justify-content-end"
        id="navbarNav"
      >
        <ul className="navbar-nav">
          {props.navLinks.map((navLink, i) => (
            <li className="nav-item mx-2" key={navLink.name}>
              <span
                className={`nav-link ${
                  navLink.name === props.currentNavLink.name && "active"
                }`}
                onClick={() => props.setCurrentNavLink(navLink)}
                data-testid={`navLink-${i}`}
              >
                {navLink.name}
              </span>
            </li>
          ))}
        </ul>
      </div>
    </div>
  </nav>;
}

export default Nav;
