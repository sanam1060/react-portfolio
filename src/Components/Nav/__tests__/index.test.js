import { render, cleanup, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Nav from "..";

const mockNavLinks = [
  {
    name: "About Me",
  },
  {
    name: "Portfolio",
  },
  {
    name: "Contact",
  },
];
const mockCurrentNavLink = jest.fn();
const mockSetCurrentNavLink = jest.fn();

afterEach(cleanup);

describe("Nav component", () => {
  it("renders", () => {
    render(
      <Nav
        navLinks={mockNavLinks}
        currentNavLink={mockCurrentNavLink}
        setCurrentNavLink={mockSetCurrentNavLink}
      ></Nav>
    );
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(
      <Nav
        navLinks={mockNavLinks}
        currentNavLink={mockCurrentNavLink}
        setCurrentNavLink={mockSetCurrentNavLink}
      ></Nav>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("contains the nav bar brand header", () => {
    render(
      <Nav
        navLinks={mockNavLinks}
        currentNavLink={mockCurrentNavLink}
        setCurrentNavLink={mockSetCurrentNavLink}
      ></Nav>
    );
    expect(screen.getByTestId("navBarHeader")).toHaveTextContent(
      "Sanam Ganjian"
    );
  });

  it("contains About Me, Portfolio, and Contact links", () => {
    render(
      <Nav
        navLinks={mockNavLinks}
        currentNavLink={mockCurrentNavLink}
        setCurrentNavLink={mockSetCurrentNavLink}
      ></Nav>
    );
    expect(screen.getByTestId("navLink-0")).toHaveTextContent("About Me");
    expect(screen.getByTestId("navLink-1")).toHaveTextContent("Portfolio");
    expect(screen.getByTestId("navLink-2")).toHaveTextContent("Contact");
  });
});
