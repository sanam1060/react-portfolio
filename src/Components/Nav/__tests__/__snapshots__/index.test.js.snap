exports[`Nav component matches the snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-md bg-light border border-secondary border-opacity-25"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand fw-bold"
        data-testid="navBarHeader"
        href="/"
      >
        Sanam Ganjian
      </a>
      <button
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-bs-target="#navbarNav"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse justify-content-end"
        id="navbarNav"
      >
        <ul
          class="navbar-nav"
        >
          <li
            class="nav-item mx-2"
          >
            <span
              class="nav-link false"
              data-testid="navLink-0"
            >
              About Me
            </span>
          </li>
          <li
            class="nav-item mx-2"
          >
            <span
              class="nav-link false"
              data-testid="navLink-1"
            >
              Portfolio
            </span>
          </li>
          <li
            class="nav-item mx-2"
          >
            <span
              class="nav-link false"
              data-testid="navLink-2"
            >
              Contact
            </span>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;